apps/06SpaceTravelToMars.js

```diff
@@ -9,7 +9,37 @@ var jobTypes = {
   programmer: 'Any Ship!'
 };
 
-// Your code here
+function CrewMember(name, job, specialSkill){
+  this.name = name;
+  this.job = job;
+  this.specialSkill = specialSkill;
+  this.ship = null;
+  this.enterShip = function (ship)  {
+      this.ship = ship;
+      ship.addCrewMember(this);
+  };
+}
+
+function Ship(name, type, ability) {
+  this.name = name;
+  this.type = type;
+  this.ability = ability;
+  this.crew = [];
+  this.missionStatement = function () {
+    for (var i = 0; i < this.crew.length; i++) {
+      var job = this.crew[i].job;
+      if (jobTypes[job] === this.type) {
+        return this.ability;
+      }
+    }
+    return "Can\'t perform a mission yet.";
```

Comment:
Good work on this! If you take a look at the last test on the bottom, you will notice that even if you check if the array is empty and return "Cant perform a mission yet," else do nothing, it will also pass.........We gotta fix it!